xonsh/built_ins.py

```diff
@@ -637,15 +637,41 @@ def run_subproc(cmds, captured=True):
             hist.last_cmd_out = output
     if not prev_is_proxy and hist.last_cmd_rtn > 0 and ENV.get('RAISE_SUBPROC_ERROR'):
         raise CalledProcessError(hist.last_cmd_rtn, aliased_cmd, output=output)
-    if captured:
+    if captured=='stdout':
         return output
+    elif captured=='object':
+        return CompletedProcess(prev_proc, output)
 
 
-def subproc_captured(*cmds):
+class CompletedProcess:
```

Comment:
Why do you create `CompletedProcess` class from scratch instead subclassing `subprocess.CompletedProcess`?

If you were subclassed `subprocess.CompletedProcess`, you wouldn't need to write `__init__` and `__getattr__` methods.

All below would be enough:

``` python
class XonshCompletedProcess(subprocess.CompletedProcess):
    def __bool__(self):
        return self.returncode == 0
    def __eq__(self, other):
        return self.returncode == other
    # Optionally, we can add `rtn` as a shorthand alias to `returncode`
    @property
    def rtn(self):
        return self.returncode
```